frontend/pages/reports/label-generator.vue

```diff
@@ -181,19 +183,29 @@
     return route(`/qrcode`, { data: encodeURIComponent(data) });
   }
 
-  function getItem(n: number): LabelData {
+  function getItem(n: number, item: { value: { name: string; location: string } } | null): LabelData {
     // format n into - seperated string with leading zeros
 
     const assetID = fmtAssetID(n);
 
     return {
       url: getQRCodeUrl(assetID),
       assetID,
-      name: "_______________",
-      location: "_______________",
+      name: item?.name ?? "_______________",
+      location: item?.location?.name ?? "_______________",
     };
   }
 
+  const { data: allFields } = await useAsyncData(async () => {
+    const { data, error } = await api.items.getAll();
```

Comment:
Not sure how to do pagination here

Reply:
Did some research into this, looks like you would do something like `getAll({ page: 1 })`

You can also define pageSize and a bunch of other stuff (your IDE should show you the whole list of possible autocompletions. However, with that said I'm not sure how we get the number of items per page and spit that out into something that the query would understand.

Reply:
I wasn't able to get my IDE to play nice with the types/autocompletes - iirc none of the globals had typing when I made this PR. I believe I was using the workspace file included in the repo so there must be something missing from that?